packages/menu-bar/src/vaadin-menu-bar-mixin.js

```diff
@@ -349,6 +359,26 @@ export const MenuBarMixin = (superClass) =>
       }
     }
 
+    /**
+     * A callback for the 'tabNavigation' property observer.
+     *
+     * @param {boolean | null} _tabNavigation
+     * @private
+     */
+    _tabNavigationChanged(_tabNavigation, overflow, container) {
+      if (overflow && container) {
+        const target = this.querySelector('[tabindex="0"]');
+        if (target) {
+          this._buttons.forEach((btn) => {
+            this._setTabindex(btn, btn === target);
+            btn.setAttribute('role', this.tabNavigation ? 'button' : 'menuitem');
```

Comment:
Could use `tabNavigation` passed to the observer instead of `this.tabNavigation` to make it a bit shorter.

Reply:
Done